ddp-workspace/projects/ddp-dsm-ui/src/app/scan/scan.component.ts

```diff
@@ -450,4 +453,40 @@ export class ScanComponent implements OnInit {
     }
     return null;
   }
+
+  public areScanPairsValid(): boolean {
+    if(!this.initialScan)
+    {
+      return this.scanPairs.length < 2;
+    }
+
+    if(this.scanPairsValue.length === 0) {
+      return true;
+    }
+    for (const pair in this.scanPairsValue) {
+      if(this.scanPairsValue[pair]['rightValue'].length !== 6 ||
+      this.scanPairsValue[pair]['leftValue'].length === 0) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  public makeScanErrorMsg(): void {
```

Comment:
This function is called too many times. Only on init, it was called 55 times.
Maybe to do something with it?
Please see the video:

https://user-images.githubusercontent.com/77500504/206187857-0aa612f3-14f0-4a46-931b-c31c710e3471.mov